### PR TITLE
[api] Move `Vernacexpr` to parsing.

### PR DESCRIPTION
**Pull request by ejgallego**
There were some spurious dependencies on the `Vernac` AST in the
pretyper, to this purpose, we slice out the bits that indeed are
defined by the lower layers, and finally place `Vernacexpr` in its
proper place.

Note that still the core AST has to be placed in the elaborator
[pretyper] and this is the main client.

### CODE REVIEW

File: interp/dumpglob.mli
Comment (herbelin) [inline interp/dumpglob.mli]: No way to preserve the abstraction given by the type (e.g. by defining it at worst in the holdall file `misctypes.ml`)? Not a big deal otherwise.

**Review by herbelin — COMMENTED** [review]
`Vernacexpr` is an interface file between `vernac` and `parsing` and as such, it made sense to me to think at it as an `intf` file. But preferring to have it close to `parsing` or `vernac` makes perfectly sense as well to me.

**Comment by ejgallego** [comment]
This PR is waiting for some other PRs to be merged, then it will be updated.

**Comment by ejgallego** [comment]
Ok, this is very close to be ready, but I want to some tweaks / improvements before it is merged.

**Review by forestjulien — APPROVED** [review]

**Comment by ejgallego** [comment]
For my taste ready to go, along with #7558

**Review by ppedrot — APPROVED** [review]
I'm fine with that.

**Comment by ppedrot** [comment]
I'm assigning this PR for the lack of any contender. This kind of architectural patches are usually not attracting much interest, so let me state out loud that if nobody complains soon this will be considered OK by default. I'll merge it in a few days thus.

**Comment by ejgallego** [comment]
Thanks @ppedrot

**Comment by ppedrot** [comment]
@ejgallego There are several failures, and if some of them look spurious, a few are definitely unclear. Can you assess the breakage?

**Review by aspiwack — APPROVED** [review]

**Comment by ejgallego** [comment]
All of them are spurious but let's indeed rerun.

**Comment by ejgallego** [comment]
> All of them are spurious but let's indeed rerun.

Well I say this as this PR has passed CI successfully many times, let's see.

**Review by gares — APPROVED** [review]

**Comment by Zimmi48** [comment]
Hum, this kind of failure:

```
==========> TESTING stm/Nijmegen_QArithSternBrocot_Zaux.v <==========
Error:
Anomaly
"Uncaught exception Failure("The spawned process did not connect back in 2.0s")."
Please report at http://coq.inria.fr/bugs/
```

seems to have become much more common recently on AppVeyor... :'(

**Comment by ejgallego** [comment]
> seems to have become much more common recently on AppVeyor... :'(

Is it a non-deterministic failure, I assume?

**Comment by ejgallego** [comment]
Let me open a question about memory use and #6859 , just in case.

**Comment by Zimmi48** [comment]
> Is it a non-deterministic failure, I assume?

Yes, see #5983 in particular.

**Comment by ejgallego** [comment]
Appveyor failed again on `par:` , that's spurious / unrelated to this AFAICS, the rest looks fine to me [pending one job in Gitlab but that one should go OK]

**Pull request merged**
